fetch/api/response/response-fragment.window.js

```diff
@@ -0,0 +1,8 @@
+promise_test(() => {
+  return fetch("./#test").then(res => {
+    assert_equals(res.url, new URL("./", location).href);
+    return fetch("../resources/redirect.py?location=/#test").then(res2 => {
+      assert_equals(res2.url, new URL("/?location=%2F&count=1", location).href);
```

Comment:
This bit took a while to understand, but I think it's correct. You really have to read redirect.py to make sense of it though, first I assumed redirect.py would simply redirect. Maybe comment "/redirect.py redirects to this funny URL, and the UA should not append a fragment"?

Reply:
Will do.